libs/application/templates/party-letter/src/fields/EndorsementList/index.tsx

```diff
@@ -5,69 +5,63 @@ import { CopyLink } from '@island.is/application/ui-components'
 
 const EndorsementList: FC<FieldBaseProps> = ({ application }) => {
+  const { lang: locale, formatMessage } = useLocale()
   const endorsementListId = (application.externalData?.createEndorsementList
     .data as any).id
-  const { formatMessage } = useLocale()
-
+  const answers = application.answers as PartyLetter
   const [searchTerm, setSearchTerm] = useState('')
-  const [endorsements, setEndorsements] = useState<Endorsement[]>()
+  const [endorsements, setEndorsements] = useState<Endorsement[] | undefined>()
   const [showWarning, setShowWarning] = useState(false)
-
-  const [getEndorsementList, { loading, error }] = useLazyQuery(
-    GET_ENDORSEMENT_LIST,
-    {
-      pollInterval: 2000,
-      onCompleted: async ({ endorsementSystemGetEndorsements }) => {
-        if (!loading && endorsementSystemGetEndorsements) {
-          const hasEndorsements =
-            !error && !loading && endorsementSystemGetEndorsements?.length
-              ? endorsementSystemGetEndorsements.length > 0
-              : false
-          const mapToEndorsementList: Endorsement[] = hasEndorsements
-            ? endorsementSystemGetEndorsements.map((x: any) => ({
-                date: x.created,
-                name: x.meta.fullName,
-                nationalId: x.endorser,
-                address: x.meta.address ? x.meta.address : '',
-                hasWarning: false,
-                id: x.id,
-              }))
-            : undefined
-          setEndorsements(mapToEndorsementList)
-        }
-      },
-    },
-  )
-
-  useEffect(() => {
-    getEndorsementList({
+  const endorsementsHook = useEndorsements(endorsementListId, true)
+  const [updateApplication] = useMutation(UPDATE_APPLICATION)
+  const updateApplicationWithEndorsements = async (
+    newEndorsements: Endorsement[],
+  ) => {
+    const updatedAnswers = {
+      ...answers,
+      endorsements: cloneDeep(newEndorsements),
+    }
+    await updateApplication({
       variables: {
         input: {
-          listId: endorsementListId,
+          id: application.id,
+          answers: {
+            ...updatedAnswers,
+          },
         },
+        locale,
       },
+    }).then(() => {
+      set(answers, 'endorsements', cloneDeep(newEndorsements))
     })
-  }, [])
+  }
+
+  useEffect(() => {
+    const mapToEndorsementList: Endorsement[] | undefined =
+      endorsementsHook && endorsementsHook.length > 0
+        ? endorsementsHook.map((x: any) => ({
+            date: x.created,
+            name: x.meta.fullName,
+            nationalId: x.endorser,
+            address: x.meta.address ? x.meta.address.streetAddress : '',
+            hasWarning: x.meta.invalidated ?? false,
+            id: x.id,
+          }))
+        : undefined
+    setEndorsements((_) => {
+      if (mapToEndorsementList && mapToEndorsementList.length > 0)
+        updateApplicationWithEndorsements(mapToEndorsementList)
+      return mapToEndorsementList
+    })
+  }, [endorsementsHook])
```

Comment:
Why are we mapping this, seems to me not doing this might reduce some complexity